usr/lib/common/pkey_utils.c

```diff
@@ -1579,7 +1579,7 @@ struct {                          \
         hash_off = 32 - hash_len;
         memcpy(param.P256.sig_r, sig, sig_len);
         memcpy(param.P256.hash + hash_off, hash, hash_len);
-        memcpy(param.P256.pub_x, ecpoint, ecpoint_len);
+        memcpy(param.P256.pub_x, ecpoint, MIN(ecpoint_len, sig_len));
```

Comment:
has sig_len be checked to be valid ? This also comes right from the user application an can thus be anything....

Reply:
Further below theris the following:

```
    case curve_ed448:
        rc = s390_kdsa(fc, edparam.buff, msg, msg_len);
        if (sig[113] != 0) {
            /* KDSA doesn't check last byte */
            rc = 1;
        }
```

Did you check that `sig_len` is at least 112 bytes ?